Ford - Add OBDC Cable Length to CarPartList

sheaduncan opened the pull request:
- [x] added new part in selfdrive/car/ford/values.py and ran `selfdrive/car/docs.py` to generate new docs

sheaduncan committed:
Update comma cable requirement for Ford CANFD

sheaduncan committed:
Adding Long_Cable to part list

incognitojam commented:
It might make more sense to add the cable to the harness definition here: https://github.com/commaai/openpilot/blob/master/selfdrive/car/docs_definitions.py#L121

sheaduncan committed:
Moving cable info into doc_definitions under CarHarness

sheaduncan committed:
Updating parts for ford_q4

sheaduncan committed:
Updating Q3 info

sheaduncan committed:
Updated MachE with USBC coupler

sheaduncan committed:
fix typo

Commit:
updating docs/cars.md

sheaduncan committed:
Add 3X angled no cable

incognitojam commented on selfdrive/car/docs_definitions.py:
```diff
@@ -117,14 +117,16 @@ class CarHarness(EnumBase):
   threex = BasePart("comma 3X", parts=[Mount.mount, Cable.right_angle_obd_c_cable_1_5ft])
   # variant of comma 3X with angled mounts
-  threex_angled_mount = BasePart("comma 3X", parts=[Mount.angled_mount_8_degrees, Cable.right_angle_obd_c_cable_1_5ft])
+  threex_angled_mount = BasePart("comma 3X", parts=[Mount.angled_mount_8_degrees, Cable.right_angle_obd_c_cable_1_5ft]
```
```suggestion
  threex_angled_mount = BasePart("comma 3X", parts=[Mount.angled_mount_8_degrees, Cable.right_angle_obd_c_cable_1_5ft])
```

sheaduncan committed:
Adding in nocable device package

sheaduncan committed:
Add missing )

Co-authored-by: Cameron Clough <cameronjclough@gmail.com>

Commit:
Generating new Docs.MD

incognitojam commented on selfdrive/car/ford/values.py:
```diff
@@ -57,9 +57,14 @@ class FordCarInfo(CarInfo):
+    elif CP.carFingerprint in (CAR.F_150_MK14, CAR.F_150_LIGHTNING_MK1):
+      self.car_parts = CarParts([Device.threex_angled_mount_nocable, harness])
+    elif CP.carFingerprint == CAR.MUSTANG_MACH_E_MK1:
+      self.car_parts = CarParts([Device.threex_angled_mount_nocable, harness, Cable.usbc_coupler])
```
Does the Mach-E need an angled mount?

sheaduncan replied:
This is my mistake, I misunderstood from a MachE user. Will update.

sheaduncan commented on selfdrive/car/docs_definitions.py:
```diff
@@ -117,14 +117,16 @@ class CarHarness(EnumBase):
   mazda = BaseCarHarness("Mazda connector")
-  ford_q3 = BaseCarHarness("Ford Q3 connector")
-  ford_q4 = BaseCarHarness("Ford Q4 connector")
+  ford_q3 = BaseCarHarness("Ford Q3 connector", parts=[Accessory.harness_box, Accessory.comma_power_v2, Cable.rj45_cable_7ft])
```
Added for consistency, is this not correct?

sheaduncan committed:
Added threex_nocable for MachE

sheaduncan committed:
Swap MachE to regular mount no cable

sheaduncan committed:
confirmed with Comma Operations all boxes include 1.5ft cable regardless of 9.5ft selection

sheaduncan committed:
Updating CarInfo

sheaduncan committed:
Update USB_Coupler for Q4 Harness

sheaduncan committed:
Remove cable from import

sheaduncan commented:
Are all these github-actions because I sent it back to draft?

sheaduncan committed:
removed ]

adeebshihadeh commented:
> Are all these github-actions because I sent it back to draft?

Not your fault. It's a bug, and I just reverted it.

sheaduncan commented:
Update:

I confirmed with the comma ops team that the 1.5ft right angle is shipped with every box. Therefore, I added the 9.5 ft cable for the F150, Lightning, and MachE, as well as the 1.5ft. Also included is the USBC coupler for use between 9.5ft cable and 1.5ft cable.

I updated the Q3 harness to be consistent with the Q4 harness. While it didn't need to be updated it makes sense to me to stay consistent in the approach.

sshane commented on selfdrive/car/docs_definitions.py:
```diff
@@ -117,8 +117,9 @@ class CarHarness(EnumBase):
   mazda = BaseCarHarness("Mazda connector")
-  ford_q3 = BaseCarHarness("Ford Q3 connector")
-  ford_q4 = BaseCarHarness("Ford Q4 connector")
+  ford_q3 = BaseCarHarness("Ford Q3 connector", parts=[Accessory.harness_box, Accessory.comma_power_v2, Cable.rj45_cable_7ft])
```
```suggestion
  ford_q3 = BaseCarHarness("Ford Q3 connector")
```

sheaduncan committed:
Disagree with the inconsistency but OK

Co-authored-by: Shane Smiskol <shane@smiskol.com>

sshane commented on selfdrive/car/ford/values.py:
```diff
@@ -57,8 +57,9 @@ class FordCarInfo(CarInfo):
   package: str = "Co-Pilot360 Assist+"
 
   def init_make(self, CP: car.CarParams):
+    # Harness in docs_definitions.py includes additional car_parts for Q3 and Q4
```
```suggestion
```

sshane committed:
Update selfdrive/car/ford/values.py
